tensorrt_llm/_torch/pyexecutor/resource_manager.py

```diff
@@ -357,6 +361,10 @@ def get_needed_resource_to_completion(self, request: LlmRequest) -> int:
     def prepare_resources(self, scheduled_batch: ScheduledRequests):
         context_batch = scheduled_batch.context_requests
         generation_batch = scheduled_batch.generation_requests
+
```

Comment:
This is additional work only relevant if `self.kv_connector_manager is not None`. Should put inside `if` statement.